app/src/components/dcCard/DCCard.tsx

```diff
@@ -0,0 +1,117 @@
+"use client";
+import React, { RefObject } from "react";
+import {
+  Card,
+  CardHeading,
+  Text,
+  CardContent,
+  Tooltip,
+  StatusBadge,
+} from "@nypl/design-system-react-components";
+import styles from "./Card.module.css";
+import { headerBreakpoints } from "../../utils/breakpoints";
+import { CollectionCardDataType } from "../../types/CollectionCardDataType";
+import { TRUNCATED_LENGTH } from "@/src/config/constants";
+import ItemCardDataType from "@/src/types/ItemCardDataType";
+interface DCCardProps {
+  cardOffset?: number[];
+  cardRef?: RefObject<HTMLDivElement>;
+  id: string;
+  isLargerThanLargeTablet: boolean;
+  slug?: string;
+  record: CollectionCardDataType | ItemCardDataType;
+}
+
+function isCollectionCardDataType(
+  record: CollectionCardDataType | ItemCardDataType
+): record is CollectionCardDataType {
+  return "numberOfDigitizedItems" in record;
+}
+
+const DCCard = ({
+  cardOffset,
+  cardRef,
+  id,
+  isLargerThanLargeTablet,
+  slug,
+  record,
+}: DCCardProps) => {
+  const truncatedTitle = record.title.length > TRUNCATED_LENGTH;
+  const isCollection = isCollectionCardDataType(record);
+  const card = (
+    <Card
+      ref={cardRef}
+      id={`card-${slug}-${id}`}
+      mainActionLink={record.url}
+      imageProps={
+        record.imageID
+          ? {
+              alt: "",
+              id: isCollection ? `image-${slug}-${id}` : `image-${id}`,
+              isLazy: true,
+              aspectRatio: "twoByOne",
+              fallbackSrc: "/noImage.png",
+              onError: (_event) =>
+                console.warn(
+                  `Card image failed to load, fallback image loaded instead. ImageURL: ${record.imageURL}`
+                ),
+              src: record.imageURL,
+            }
+          : {
+              alt: "",
+              id: `no-image-${id}`,
+              isLazy: true,
+              aspectRatio: "twoByOne",
+              src: "/noImage.png",
+            }
+      }
+    >
+      <CardContent>
+        {isCollection && record.containsOnSiteMaterials && (
```

Comment:
Based on the interfaces declared, the `CollectionCardDataType` interface is the only type that has `containsOnSiteMaterials`. Can you include this check in the `isCollectionCardDataType` type guard? In other words, is it true that a collection contains both `containsOnSiteMaterials` and `numberOfDigitizedItems`? Or is there another possible type that contains `numberOfDigitizedItems` but not `containsOnSiteMaterials` and is not an `ItemCardDataType`?

Reply:
My thinking here was that ultimately items, as they're represented in DCF, will have some kind of `isOnsite` field. I doubt it would be the same name but the idea is similar enough that I didn't want to include it in the type guard. `numberOfDigitizedItems` is clearly never going to be a field on an item, so I think that's the only safe bet so far– not to say this type guard couldn't be built up once we have a definitive `Item` type (as you mention below)